rp2040-hal/src/timer.rs

```diff
@@ -172,9 +172,9 @@ pub trait Alarm {
     /// reason, the minimum time that this function accepts is `10.microseconds()`
     ///
     /// [enable_interrupt]: #method.enable_interrupt
-    fn schedule<TIME: Into<Microseconds>>(
```

Comment:
Trying to use this I find the ergonomics a bit weird (might be me)...

So in converting I went from: 
`alarm0.schedule(TIME_TICK_MS.microseconds() * 1000);`
to
`alarm0.schedule(TIME_TICK_MS.micros() * 1000);`
which gave me an error because schedule is generic over the input duration NOM and DENOM. Which forces met to do:
`alarm0.schedule::<1, 1000_000>(TIME_TICK_MS.micros() * 1000)`

Since we convert to millis below, does it not make sense to just accept MillisDurationU32 since the caller can easily convert?